src/sql/workbench/browser/designer/designer.ts

```diff
@@ -836,6 +850,48 @@ export class Designer extends Disposable implements IThemable {
 					headerRowHeight: TableHeaderRowHeight,
 					editorLock: new Slick.EditorLock()
 				});
+				let buttonColumn: Slick.Column<Slick.SlickData>[] = [];
+				if (tableProperties.canInsertRows) {
+					// Add move context menu actions
+					this._register(table.onContextMenu((e) => {
+						this.openContextMenu(table, e, propertyPath, view);
+					}));
+				}
+				if (tableProperties.canMoveRows) {
+					// Add row move drag and drop
+					const moveRowsPlugin = new RowMoveManager({
+						cancelEditOnDrag: true,
+						id: 'moveRow',
+						iconCssClass: Codicon.grabber.classNames,
+						title: localize('designer.moveRowText', 'Move Row'),
+						width: 20,
+						resizable: false,
+						isFontIcon: true,
+						behavior: 'selectAndMove'
+					});
+					table.registerPlugin(moveRowsPlugin);
+					moveRowsPlugin.onBeforeMoveRows.subscribe((e: Slick.EventData, data: RowMoveOnDragEventData) => {
+						// no point in moving before or after itself
+						if (data.rows[0] === data.insertBefore || data.rows[0] === data.insertBefore - 1) {
+							e.stopPropagation();
+							return;
+						}
+					});
+					moveRowsPlugin.onMoveRows.subscribe((e: Slick.EventData, data: RowMoveOnDragEventData) => {
+						const row = data.rows[0];
+						this.handleEdit({
+							type: DesignerEditType.Move,
+							path: [...propertyPath, row],
+							source: view,
+							value: data.insertBefore
```

Comment:
As per our discussion, the value need to be the new index of the column. It should be:
1) `data.insertBefore - 1` if the destination is after current position. 
2) `data.insertBefore` if the destination is before current position

Reply:
alright let me try with your updated change in STS